test/functional/p2p_disabletx.py

```diff
@@ -0,0 +1,175 @@
+        peer = self.nodes[0].add_outbound_p2p_connection(P2PInterface(disabletx=True), p2p_idx=0, connection_type='outbound-full-relay', expect_disconnect=True)
+        peer.wait_for_disconnect()
+
+    def getdata_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending getdata(tx) or getdata(merkleblock) after disabletx')
+        for inv_type in (MSG_TX, MSG_TX | MSG_WITNESS_FLAG, MSG_WTX, MSG_FILTERED_BLOCK):
+            wtxidrelay = False
+            if inv_type == MSG_WTX:
+                wtxidrelay = True
+            peer = self.nodes[0].add_p2p_connection(P2PInterface(wtxidrelay=wtxidrelay, disabletx=True))
+            peer.sync_with_ping()
+
+            peer.send_message(msg_getdata(inv=[CInv(t=inv_type, h=self.tip_hash)]))
+            peer.wait_for_disconnect()
+
+    def inv_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending inv(tx) after disabletx')
+        for inv_type in (MSG_TX, MSG_WTX):
+            wtxidrelay = False
+            if inv_type == MSG_WTX:
+                wtxidrelay = True
+            peer = self.nodes[0].add_p2p_connection(P2PInterface(wtxidrelay=wtxidrelay, disabletx=True))
+            peer.sync_with_ping()
+
+            peer.send_message(msg_inv(inv=[CInv(t=inv_type, h=0xdeadbeef)]))
+            peer.wait_for_disconnect()
+
+    def tx_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending a tx after disabletx')
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface(disabletx=True))
+        peer.sync_with_ping()
+
+        # Construct a dummy transaction. Inputs will be missing so no
+        # validation will occur (this is easier than constructing a valid
+        # transaction, which should be unnecessary for this test).
+        tx = CTransaction()
+        tx.vin = [CTxIn(outpoint=COutPoint(0xdeadbeef, 0))]
+        tx.vout = [CTxOut()]
+
+        peer.send_message(msg_tx(tx))
+        peer.wait_for_disconnect()
+
+    def bloom_filter_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending a bip37 filter* message after disabletx')
+
+        for msg in (msg_filterload(), msg_filterclear(), msg_filteradd(data=b'00')):
+            peer = self.nodes[0].add_p2p_connection(P2PInterface(disabletx=True))
+            peer.sync_with_ping()
+
+            peer.send_message(msg)
+            peer.wait_for_disconnect()
+
+    def mempool_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending a mempool message after disabletx')
+
+        peer = self.nodes[0].add_p2p_connection(P2PInterface(disabletx=True))
+        peer.sync_with_ping()
+
+        peer.send_message(msg_mempool())
+        peer.wait_for_disconnect()
+
+    def notfound_after_disabletx_test(self):
+        self.log.info('Check for disconnection when sending a notfound(tx) message after disabletx')
+
+        for inv_type in (MSG_TX, MSG_WTX):
```

Comment:
Note, I wonder if we can send back a notfound(MSG_TX | MSG_WITNESS_FLAG) as those messages are replied to getdatas.